src/build/help/help.xml

```diff
@@ -1087,6 +1097,22 @@
                         <example>~/.ssh/id_ed25519.pub</example>
                     </config-key>
 
+                    <config-key id="repo-sftp-strict-host-key-check" name="SFTP host key check type">
+                        <summary>SFTP host key check type.</summary>
+
+                        <text>
+                            <p>The following SFTP host key check types are supported:</p>
+                            <list>
+                                <list-item><id>yes</id> - <backrest/> will never automatically add host keys to the ~/.ssh/known_hosts file, and refuses to connect to hosts whose host key has changed or is not found in the known hosts files. This option forces the user to manually add all new hosts</list-item>
+                                <list-item><id>accept-new</id> - <backrest/> will automatically add new host keys to the user's known hosts file, but will not permit connections to hosts with changed host keys</list-item>
+                                <list-item><id>no</id> - <backrest/> will automatically add new host keys to the user's known hosts file and allow connections to hosts with changed host keys to proceed</list-item>
+                            </list>
```

Comment:
Wow, this seems a bit crazy. Why would you add the host keys at all if you are just ignoring them? Is this how OpenSSH operates?

Reply:
I think this defines the restrictions https://github.com/openssh/openssh-portable/blob/b6b49130a0089b297245ee39e769231d7c763014/sshconnect.c#L1298

Reply:
Well, this kind of makes it look like we should not be updating host keys in this case: https://github.com/openssh/openssh-portable/blob/b6b49130a0089b297245ee39e769231d7c763014/sshconnect.c#L1342?

Reply:
But this does `If this flag is set to “accept-new” then ssh will automatically add new host keys to the user known hosts files, but will not permit connections to hosts with changed host keys.  If this flag is set to “no” or “off”, ssh will automatically add new host keys to the user known hosts files and allow connections to hosts with changed hostkeys to proceed, subject to some restrictions.` Seems dumb as bricks to me, and I wonder what the `restrictions` are.

Reply:
I think I don't care what OpenSSH does here -- no means no. Perhaps it's OK for OpenSSH to muck about with these files but I don't think we should.

Reply:
I'm fine with that.  I agree it seems pretty crazy and I'm not sure why OpenSSH does it that way.  I would prefer to know why they do it that way, but we don't and so I'm thinking that we go ahead and do what we think makes the most sense here.

Reply:
> Well, this kind of makes it look like we should not be updating host keys in this case: https://github.com/openssh/openssh-portable/blob/b6b49130a0089b297245ee39e769231d7c763014/sshconnect.c#L1342?

Just to note.
UpdateHostKeys is different from add a new hostkey. 
 UpdateHostKeys                                                                                                                     
 Specifies whether ssh(1) should accept notifications of additional hostkeys from the server sent after authentication has completed and add them to UserKnownHostsFile.  The argument must be yes, no or ask.  This option allows learning alternate hostkeys for a server and supports graceful key rotation by allowing a server to send replacement public keys before old ones are removed.  Additional hostkeys are only accepted if the key used to authenticate the host was already trusted or explicitly accepted by the user.                                        
                                                                                                                                    
 UpdateHostKeys is enabled by default if the user has not overridden the default UserKnownHostsFile setting, otherwise UpdateHostKeys will be set to ask.
                                                                                                                                    
 If UpdateHostKeys is set to ask, then the user is asked to confirm the modifications to the known_hosts file.  Confirmation is currently incompatible with ControlPersist, and will be disabled if it is enabled.
                                                                                                                                    
 Presently, only sshd(8) from OpenSSH 6.8 and greater support the "hostkeys@openssh.com" protocol extension used to inform the client of all the server's hostkeys.

Re adding new host keys.
For  case HOST_NEW:

https://github.com/openssh/openssh-portable/blob/b6b49130a0089b297245ee39e769231d7c763014/sshconnect.c#L1191